bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -133,20 +211,269 @@ logging.info("calculating ifar/fap values")
 if fore_locs.sum() > 0:
     ifar = background_time / (fnlouder + 1)
     fap = 1 - numpy.exp(- coinc_time / ifar)
-    f['foreground/ifar'] = sec_to_year(ifar)
-    f['foreground/fap'] = fap
+    f['foreground_h' + h_iter_str + '/ifar'] = sec_to_year(ifar)
+    f['foreground_h' + h_iter_str + '/fap'] = fap
 
     ifar_exc = background_time_exc / (fnlouder_exc + 1)
     fap_exc = 1 - numpy.exp(- coinc_time_exc / ifar_exc)
     f['foreground/ifar_exc'] = sec_to_year(ifar_exc)
     f['foreground/fap_exc'] = fap_exc
 else:
-    f['foreground/ifar'] = numpy.array([])
-    f['foreground/fap'] = numpy.array([])
+    f['foreground_h' + h_iter + '/ifar'] = numpy.array([])
+    f['foreground_h' + h_iter + '/fap'] = numpy.array([])
     f['foreground/ifar_exc'] = numpy.array([])
     f['foreground/fap_exc'] = numpy.array([])
 
-if 'name' in d.attrs:
-    f.attrs['name'] = d.attrs['name']
+if 'name' in all_trigs.attrs:
+    f.attrs['name'] = all_trigs.attrs['name']
+
+# Incorporate hierarchical removal for any other loud triggers
+logging.info("Beginning hierarchical removal of foreground triggers.")
+
+# Step 1: Check to see if there are triggers that are louder
+#         then all of the inclusive background.
+
+# fnlouder is the number of background events that are louder than the
+# foreground trigger. So if a foreground trigger is louder than all of the
+# inclusive background, fnlouder = 0.
+
+loud_t_ind = numpy.where(fnlouder == 0)[0]
+
+# Create arrays to save far and fap of hierarchically removed
+# foreground triggers
+
+save_ifar_fore_trig = numpy.array([], dtype=float)
+save_fap_fore_trig = numpy.array([], dtype=float)
+save_these_fore_ind = numpy.array([], dtype=int)
+
+# Use a copy of the original triggers befpre hierarchical removal
+# to regenerate hierarchically removed foreground triggers
+copy_trigs = pycbc.io.StatmapData(all_trigs.data, all_trigs.seg,
+                                  all_trigs.attrs)
+
+# Step 2 : Loop until we don't have to hierarchically remove
+#          anymore.
+
+while len(loud_t_ind) > 0:
+    # If the user wants to stop doing hierarchical removals after a set
+    # number of iterations then break when the h_iterations gets to that
+    # number
+
+    if (h_iterations == args.max_hier_removal): 
+        logging.warn("Hierarchical removal can continue but user has set"
```

Comment:
It's not needed, I suppose if one looks at the plot it's fairly obvious when more hierarchical removals need to be done. I'll remove it.